### PR TITLE
PMM-10799-11465-11544-11577-11729 QAN explain table

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
https://jira.percona.com/browse/PMM-10799
https://jira.percona.com/browse/PMM-11465
https://jira.percona.com/browse/PMM-11544
https://jira.percona.com/browse/PMM-11577
https://jira.percona.com/browse/PMM-11729

More PRs are together, because they are related.

Link to the Feature Build: SUBMODULES-0

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

**Review by rishat-ishbulatov — APPROVED** [review]

File: agent/runner/actions/mysql_explain_action.go [hunk 1]
```diff
@@ -73,13 +73,23 @@ func (a *mysqlExplainAction) Type() string {
 
 // Run runs an Action and returns output and error.
 func (a *mysqlExplainAction) Run(ctx context.Context) ([]byte, error) {
+	if a.params.Query == "" {
+		return nil, fmt.Errorf("query to EXPLAIN is empty")
```
Comment (artemgavrilov) [inline agent/runner/actions/mysql_explain_action.go:77]:
```suggestion
		return nil, errors.New("query to EXPLAIN is empty")
```

File: agent/runner/actions/mysql_explain_action.go [hunk 2]
```diff
@@ -73,13 +73,23 @@ func (a *mysqlExplainAction) Type() string {
...
+	}
+
+	// Workaround for bug in our MySQL parser if there is keyword "IN" in query.
+	// TODO In future it should be fixed on parser side.
+	a.params.Query = strings.ReplaceAll(a.params.Query, "...", "?")
+
+	// Explain is supported only for DML queries.
+	// https://dev.mysql.com/doc/refman/8.0/en/using-explain.html
+	if !isDMLQuery(a.params.Query) {
+		return nil, fmt.Errorf("functionality EXPLAIN is supported only for DML queries (SELECT, INSERT, UPDATE, DELETE, REPLACE)")
```
Comment (artemgavrilov) [inline agent/runner/actions/mysql_explain_action.go:87]:
```suggestion
		return nil, errors.New("functionality EXPLAIN is supported only for DML queries (SELECT, INSERT, UPDATE, DELETE, REPLACE)")
```

File: agent/runner/actions/query_transform.go [hunk 3]
```diff
@@ -47,36 +48,41 @@ are needed and the pmm user is a not privileged user.
 This function converts DML queries to the equivalent SELECT to make
 it able to explain DML queries on older MySQL versions and for unprivileged users.
 */
-func dmlToSelect(query string) string {
+func dmlToSelect(query string) (string, bool) {
```
Comment (artemgavrilov) [inline agent/runner/actions/query_transform.go:51]: Can you please add comment with explanation of what this bool stands for?

**Review by artemgavrilov — APPROVED** [review]

File: agent/runner/actions/query_transform.go (hunk 3 above)
Comment (JiriCtvrtka) [reply agent/runner/actions/query_transform.go:52]: Sure. I added comment to func:
// dmlToSelect returns query converted to select and boolean, if conversion were needed.

Take a look if its clear, thank you.

File: agent/runner/actions/mysql_show_table_status_action.go [hunk 4]
```diff
@@ -65,7 +67,20 @@ func (a *mysqlShowTableStatusAction) Run(ctx context.Context) ([]byte, error) {
...
 	defer tlshelpers.DeregisterMySQLCerts()
 
-	rows, err := db.QueryContext(ctx, "SHOW /* pmm-agent */ TABLE STATUS WHERE Name = ?", a.params.Table)
+	table := a.params.Table
+	if containsDB := strings.Contains(a.params.Table, "."); containsDB {
+		split := strings.Split(a.params.Table, ".")
+		if len(split) > 1 {
+			useQuery := fmt.Sprintf("USE /* pmm-agent */ %s;", split[0])
+			table = split[1]
+			_, err = db.QueryContext(ctx, useQuery)
```
[the remark this answers is not in the document]
Comment (JiriCtvrtka) [reply agent/runner/actions/mysql_show_table_status_action.go]: db will be closed because on beginning there is close in defer.

**Pull request merged**